examples/mnist_tfrecord.py

```diff
@@ -0,0 +1,158 @@
+
+sess = K.get_session()
+
+batch_size = 128
+batch_shape = (batch_size, 28, 28, 1)
+steps_per_epoch = 469
+epochs = 78
+classes = 10
+
+# The capacity variable controls the maximum queue size
+# allowed when prefetching data for training.
+capacity = 10000
+
+# min_after_dequeue is the minimum number elements in the queue
+# after a dequeue, which ensures sufficient mixing of elements.
+min_after_dequeue = 3000
+
+# If `enqueue_many` is `False`, `tensors` is assumed to represent a
+# single example.  An input tensor with shape `[x, y, z]` will be output
+# as a tensor with shape `[batch_size, x, y, z]`.
+#
+# If `enqueue_many` is `True`, `tensors` is assumed to represent a
+# batch of examples, where the first dimension is indexed by example,
+# and all members of `tensors` should have the same size in the
+# first dimension.  If an input tensor has shape `[*, x, y, z]`, the
+# output will have shape `[batch_size, x, y, z]`.
+enqueue_many = True
+
+data = mnist.load_mnist()
+x_train_batch, y_train_batch = tf.train.shuffle_batch(
+    tensors=[data.train.images, data.train.labels.astype(np.int32)],
+    batch_size=batch_size,
+    capacity=capacity,
+    min_after_dequeue=min_after_dequeue,
+    enqueue_many=enqueue_many,
+    num_threads=8)
+
+x_train_batch = tf.cast(x_train_batch, tf.float32)
+x_train_batch = tf.reshape(x_train_batch, shape=batch_shape)
+
+y_train_batch = tf.cast(y_train_batch, tf.int32)
+y_train_batch = tf.one_hot(y_train_batch, classes)
+
+x_batch_shape = x_train_batch.get_shape().as_list()
+y_batch_shape = y_train_batch.get_shape().as_list()
+
+x_train_input = layers.Input(tensor=x_train_batch, batch_shape=x_batch_shape)
+x_train_out = cnn_layers(x_train_input)
+train_model = Model(inputs=x_train_input, outputs=x_train_out)
+
+cce = objectives.categorical_crossentropy(y_train_batch, x_train_out)
+train_model.add_loss(cce)
+
+# Do not pass the loss directly to model.compile()
+# because it is not yet supported for Input Tensors.
+train_model.compile(optimizer='rmsprop',
+                    loss=None,
+                    metrics=['accuracy'])
+train_model.summary()
+
+coord = tf.train.Coordinator()
+threads = tf.train.start_queue_runners(sess, coord)
+train_model.fit(epochs=epochs,
+                steps_per_epoch=steps_per_epoch)
+
+train_model.save_weights('saved_wt.h5')
+
+coord.request_stop()
+coord.join(threads)
+K.clear_session()
+
+# Second Session to test loading trained model without tensors
+x_test = np.reshape(data.validation.images, (data.validation.images.shape[0], 28, 28, 1))
+y_test = data.validation.labels
+x_test_inp = layers.Input(batch_shape=(None,) + (x_test.shape[1:]))
+test_out = cnn_layers(x_test_inp)
+test_model = Model(inputs=x_test_inp, outputs=test_out)
+
+test_model.load_weights('saved_wt.h5')
+test_model.compile(optimizer='rmsprop', loss='categorical_crossentropy', metrics=['accuracy'])
+test_model.summary()
+
+loss, acc = test_model.evaluate(x_test, np_utils.to_categorical(y_test), classes)
```

Comment:
The simplicity of the test hides the lack of underlying implementation.

In this test there is no label `y` to evaluate against. You'll see in the `_test_loop()` the [output of f(ins) is not saved](https://github.com/fchollet/keras/pull/7113/files#diff-299cfd5886683a4b012f286403769fc1R1248) in the `steps` case, which is ok for this PR as it was originally conceived. 

This seemingly simple missing component actually requires some implementation to work.  I'm not sure if supplying a Tensor as the parameter `y` will give results in the format described by `#Returns`. Perhaps simply retaining a list of the outputs is OK like in [_step_loop()](https://github.com/fchollet/keras/pull/6928/commits/3e13d6c0d82f37cb003e48c440f212a18e3347ee#diff-299cfd5886683a4b012f286403769fc1R1292)? (the link is to the #6928 version of the changes I reverted here)

I'm not sure about indexing correctly to calculate the summaries for each output in this case, e.g. something equivalent to `outs[i][batch_start:batch_end] = batch_out` in `_predict_loop()` or in `_test_loop` there is ` outs[i] += batch_out * len(batch_ids)` and then `outs[i] /= samples`.